components/core/src/clp/string_utils/string_utils.hpp

```diff
@@ -140,6 +144,53 @@ bool convert_string_to_int(std::string_view raw, integer_t& converted) {
         return result.ec == std::errc();
     }
 }
+
+/**
+ * This class implements a validating UTF-8 parser which takes as input valid or nearly-valid UTF-8
+ * and outputs valid UTF-8, handling invalid UTF-8 characters according to some policy.
+ */
+class ValidatingUtf8Parser {
+public:
+    enum class InvalidUtf8Policy : uint8_t {
+        SubstituteReplacementCharacter,
+        ReturnError
+    };
```

Comment:
How about:
```suggestion
    enum class ErrorHandlingPolicy : uint8_t {
        Strict,
        Replace
    };
```
- Since this class is inside this utf8 validation class, we don't have to duplicate "Utf8" in its name
- `SubstituteReplacementCharacter` -> `Replace`, `ReturnError` -> `Strict` to mirror Python's error handling policy names ([here](https://docs.python.org/3/library/codecs.html#error-handlers)), which might be more concise.

Open to other naming plans if u have any